src/coffea/nanoevents/schemas/nanoaod.py

```diff
@@ -206,6 +212,24 @@ def _build_collections(self, field_names, input_contents):
                     branch_forms["n" + name]
                 )
 
+        # Check the presence of the event_ids
+        missing_event_ids = [
+            event_id for event_id in self.event_ids if not hasattr(self, event_id)
+        ]
+        if missing_event_ids:
+            if self.error_missing_event_ids:
+                raise Exception(
+                    f"event_ids {missing_event_ids} is/are missing. "
+                    "These fields can be needed to distinguish problematic "
+                    "data collection sub-eras or to do an analysis cross-check.\n"
+                    "To prevent this error turn error_missing_event_ids to False"
```

Comment:
I think these 3 lines can read as follows (@andrzejnovak @lgray @nsmith- crosscheck me on verbosity):
"The event ID fields {self.event_ids} are necessary to perform sub-run identification (e.g. for corrections and sub-dividing data during different detector conditions), to cross-validate MC and Data (i.e. matching events for comparison), and to generate event displays. It's advised to never drop these branches from the dataformat.

This error can be demoted to a warning by setting the class level variable ``error_missing_event_ids`` to False."